src/main/java/com/bobocode/service/UserService.java

```diff
@@ -0,0 +1,88 @@
+package com.bobocode.service;
+
+import com.bobocode.dto.Note;
+import com.bobocode.dto.User;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import org.springframework.util.StringUtils;
+import org.springframework.web.client.HttpClientErrorException;
+
+import java.util.*;
+import java.util.concurrent.ConcurrentHashMap;
+
+@Service
+public class UserService {
+    private final Map<String, User> userById = new ConcurrentHashMap<>();
+
+    public Collection<User> findAll() {
+        return userById.values();
+    }
+
+    public User getUserById(String id) {
+        checkId(id);
+        return Optional.ofNullable(userById.get(id))
+                .orElseThrow(() -> new HttpClientErrorException(HttpStatus.NOT_FOUND, "User by id:%s not found".formatted(id)));
+    }
+
+    public void createUser(User user) {
+        if (user.getId() != null) {
+            throw new HttpClientErrorException(HttpStatus.BAD_REQUEST, "User id must be empty");
+        }
+        user.setId(generateId());
+        userById.put(user.getId(), user);
+    }
+
+    public void deleteUserById(String id) {
+        checkId(id);
+        userById.remove(id);
+    }
+
+    public void createUserNote(String id, Note note) {
+        if (note.getId() != null) {
+            throw new HttpClientErrorException(HttpStatus.BAD_REQUEST, "Note id must be empty");
+        }
+
+        if (note.getUserId() != null) {
+            throw new HttpClientErrorException(HttpStatus.BAD_REQUEST, "User id must be empty");
+        }
+
+        User user = getUserById(id);
+        note.setId(generateId());
+        note.setUserId(id);
+
+        if (user.getNotes() == null) {
+            user.setNotes(new ArrayList<>());
+        }
+
+        user.getNotes().add(note);
```

Comment:
you can create a helper method in User class addNote()
and encapsulate all this logic there
`note.setId(generateId());
        note.setUserId(id);

        if (user.getNotes() == null) {
            user.setNotes(new ArrayList<>());
        }

        user.getNotes().add(note);`

Reply:
Nice idea! I'll do that, thanks!

Reply:
Fixed